db/seeds.rb

```diff
@@ -6,18 +6,31 @@
 #
 #   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
 #   Character.create(name: 'Luke', movie: movies.first)
-admin_email = ENV.fetch('ADMIN_USER', 'admin@barong.io')
 
-admin = Account.create(email: admin_email, password: SecureRandom.hex(20), level: 1, role: 'admin', confirmed_at: Time.now)
+seed_file =  File.join(Rails.root, 'config', 'seed.yml')
 
-puts 'Admin credentials: %s' % [admin.password]
-
-if File.exist?([Rails.root, '/config', '/seed.yml'].join)
-  seed_file = File.join(Rails.root, 'config', 'seed.yml')
+if File.exist?(seed_file)
   seed = YAML.load(File.open(seed_file))
 
-  app_seed = seed['oauth_application']
-  application = Doorkeeper::Application.new(skipauth: true, name: app_seed['oauth_app_name'], redirect_uri: app_seed['oauth_callback_url'])
-  application.save!
-end
+  admin = Account.create(email: seed['admin']['email'], password: SecureRandom.hex(20), level: 1, role: 'admin', confirmed_at: Time.now)
+
+  puts 'Admin credentials: %s' % [admin.password]
 
+  # Create applications from seed
+  seed['applications'].each do |app|
+    unless app['uid'].nil? && app['secret'].nil?
+      result = Doorkeeper::Application.new(skipauth: app['skipauth'], name: app['name'], redirect_uri: app['callback_url'], uid: app['uid'], secret: app['secret'])
+    else
+      result = Doorkeeper::Application.new(skipauth: app['skipauth'], name: app['name'], redirect_uri: app['callback_url'])
+    end
```

Comment:
why you don't simply run
Doorkeeper::Application.new(app)

Reply:
true :sweat_smile: